### PR TITLE
fix: Stop exporting `Portal` component from react-components

### PR DESCRIPTION
**Pull request by ling1726**
The component is not a part of any design spec, it's simply an internal
component used to position other components like Tooltip, Menu, Popover
out of DOM order.

### CODE REVIEW

**Review by miroslavstastny — APPROVED** [review]

**Review by layershifter — APPROVED** [review]

**Comment by spmonahan** [comment]
> This is a breaking change, so we should add this to the list of breaking changes CC: @spmonahan
> 
> @ling1726 I assume we don't think any partners are using this directly?

What are the consequences of not removing this?

**Comment by ling1726** [comment]
> > This is a breaking change, so we should add this to the list of breaking changes CC: @spmonahan
> > @ling1726 I assume we don't think any partners are using this directly?
> 
> What are the consequences of not removing this?

@spmonahan  it's always easier to release something when there is a legitimate need for it, otherwise we risk inappropriate usage of internals which will increase our API surface for maintenance

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/3321df44cdc17b93c110019f7365bc80f0004e1b'>3321df44cdc17b93c110019f7365bc80f0004e1b</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23178/perf-test-react-components/Avatar-mount_base.html">1255</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23178/perf-test-react-components/Avatar-mount.html">1284</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23178/perf-test-react-components/Button-mount_base.html">791</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23178/perf-test-react-components/Button-mount.html">788</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23178/perf-test-react-components/FluentProvider-mount_base.html">2343</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23178/perf-test-react-components/FluentProvider-mount.html">2361</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23178/perf-test-react-components/FluentProviderWithTheme-mount_base.html">399</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23178/perf-test-react-components/FluentProviderWithTheme-mount.html">451</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23178/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">362</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23178/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">370</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23178/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">466</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23178/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">443</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23178/perf-test-react-components/MakeStyles-mount_base.html">2005</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23178/perf-test-react-components/MakeStyles-mount.html">2060</a></td>
            <td>50000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by JustSlone** [comment]
We should also remove portal docs from the storybook as well, right? Fine to be in a separate PR since that's not a breaking change. 

https://react.fluentui.dev/?path=/docs/components-portal--default

File: change/@fluentui-react-components-dba74903-4ac6-42e6-bd6b-d63c496ea25c.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "prerelease",
+  "comment": "fix: Stop exporting `Portal` component from react-components",
```
Comment (spmonahan) [inline change/@fluentui-react-components-dba74903-4ac6-42e6-bd6b-d63c496ea25c.json:3]:
```suggestion
  "comment": "BREAKING: fix: Stop exporting `Portal` component from react-components",
```

**Pull request merged**